worker/src/lib.rs

```diff
@@ -170,38 +160,59 @@ async fn linkup_request_handler<'a, S: StringStore>(
     let mut cf_resp =
         convert_reqwest_response_to_cf(response, &additional_response_headers()).await?;
 
-    cf_resp = set_cached_req(&req, cf_resp, config.cache_routes).await?;
+    if is_cacheable_request(&req, &config) {
+        cf_resp = set_cached_req(&req, cf_resp, session_name).await?;
+    }
 
     Ok(cf_resp)
 }
 
-#[event(fetch)]
-pub async fn main(req: Request, env: Env, _ctx: worker::Context) -> Result<Response> {
-    // Optionally, get more helpful error messages written to the console in the case of a panic.
-    utils::set_panic_hook();
-
-    let kv = match env.kv("LINKUP_SESSIONS") {
-        Ok(kv) => kv,
-        Err(e) => return plaintext_error(format!("Failed to get KV store: {}", e), 500),
-    };
-
-    let string_store = CfWorkerStringStore::new(kv);
+fn is_cacheable_request(req: &Request, config: &Session) -> bool {
+    if req.method() != Method::Get {
+        return false;
+    }
 
-    let sessions = SessionAllocator::new(&string_store);
+    if config.session_token == PREVIEW_SESSION_TOKEN {
+        return true;
+    }
 
-    if let Ok(Some(upgrade)) = req.headers().get("upgrade") {
-        if upgrade == "websocket" {
-            return linkup_ws_handler(req, &sessions).await;
+    if let Some(routes) = &config.cache_routes {
+        let path = req.path();
+        if routes.iter().any(|route| route.is_match(&path)) {
+            return true;
         }
     }
 
-    if req.method() == Method::Post && req.path() == "/linkup" {
-        return linkup_session_handler(req, &sessions).await;
-    }
+    false
+}
 
-    if req.method() == Method::Post && req.path() == "/preview" {
-        return linkup_preview_handler(req, &sessions).await;
+fn get_cache_key(req: &Request, session_name: &String) -> Result<String> {
+    let mut cache_url = req.url()?.clone();
+    let curr_domain = cache_url.domain().ok_or("example.com")?;
+    cache_url.set_host(Some(&format!("{}.{}", session_name, curr_domain)))?;
+
+    Ok(cache_url.to_string())
+}
+
+async fn get_cached_req(req: &Request, session_name: &String) -> Result<Option<Response>> {
+    let cache_key = get_cache_key(req, session_name)?;
+    Cache::default().get(cache_key, false).await
+}
```

Comment:
**_thought (non-blocking): 💭_** 
Not fully aware of the use case, but my initial thought is that maybe cache key shouldn't be a failable operation?
Could possibly be fine with being an Option that returns None in case of failure 🤔  But again, I'm not fully aware of the feature, so maybe we do want to hard fail the request in case is a cacheable request and the key cannot be resolved.

Reply:
Yeah, I agree with you here. Had one crack at removing the Result type already but I wasn't super sure on the defaults, so skipped it for now.

Reply:
Want me to take a quick stab at it? 👀

Reply:
Yes pls 🤗

Reply:
What do you think of 0fb583d6fa694f03435121a2110409a24658eb95?

Reply:
I like it I love it!